### PR TITLE
change module name to index and remove npmignore

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,10 +2,17 @@
...
   "description": "Performant, flexible and extensible forms library for React Hooks",
   "version": "6.0.0-rc.5",
-  "main": "dist/react-hook-form.js",
-  "module": "dist/react-hook-form.es.js",
+  "main": "dist/index.js",
+  "main:ie": "dist/index.ie11.js",
+  "umd:main": "dist/index.umd.js",
+  "unpkg": "dist/index.umd.min.js",
+  "module": "dist/index.esm.js",
+  "module:min": "dist/index.min.esm.js",
```
Comment (keiya01) [inline package.json:10]: Are `main:ie` and `umd:main` and so on used in only rollup config?

**Review by keiya01 — COMMENTED** [review]
I have a question. Thank you.

File: package.json (hunk 1 above)
Comment (bluebill1049) [reply package.json:10]: yes, is there any issues?

Comment (keiya01) [reply package.json:10]: There are no issues. I just wanted to hear the reason :)

**Review by keiya01 — APPROVED** [review]
LGTM!

File: package.json (hunk 1 above)
Comment (bluebill1049) [reply package.json:10]: `main:ie` is for IE users, umd for users who use `umd`.

Comment (keiya01) [reply package.json:10]: Exactly, I wanted to confirm while these variables are used only in the config or not.

**Comment by ryansully** [comment]
I noticed that https://react-hook-form.com/faqs#BrowserSupport still points to the old path. Should this be `import { useForm } from 'react-hook-form/dist/index.ie11'` instead?

**Comment by bluebill1049** [comment]
@ryansully send a PR please 🙏

**Comment by bluebill1049** [comment]
send a PR please @ryansully

**Pull request merged**